src/Utils/D3D.cpp

```diff
@@ -90,12 +90,21 @@ namespace Util
 	ID3D11DeviceChild* CompileShader(const wchar_t* FilePath, const std::vector<std::pair<const char*, const char*>>& Defines, const char* ProgramType, const char* Program)
 	{
 		auto& device = State::GetSingleton()->device;
-
 		// Build defines (aka convert vector->D3DCONSTANT array)
 		std::vector<D3D_SHADER_MACRO> macros;
+		std::string str;
+		std::wstring path{ FilePath };
+		std::transform(path.begin(), path.end(), std::back_inserter(str), [](wchar_t c) {
+			return (char)c;
+		});
```

Comment:
Why are this chunk of code moved here

Reply:
so I can use the converted str in the logger

Reply:
Ok, maybe replace it with `Util::WStringToString`

Reply:
done 👍